kubeconfig_test.go

```diff
@@ -93,3 +100,213 @@ func TestKubeConfig_getRESTConfigFromSecret(t *testing.T) {
+						Cluster: KubeconfigCluster{
+							Server: "https://10.142.5.51:8443",
+						},
+					},
+				},
+				Users: []KubeconfigUser{
+					{
+						Name: "minikube",
+						User: KubeconfigUserKeyPair{
+							ClientCertificateData: "/workdir/.minikube/client.crt",
+							ClientKeyData:         "/workdir/.minikube/client.key",
+						},
+					},
+				},
+				Contexts: []KubeconfigNamedContext{
+					{
+						Name: "minikube",
+						Context: KubeconfigContext{
+							Cluster: "minikube",
+							User:    "minikube",
+						},
+					},
+				},
+				CurrentContext: "minikube",
+			},
+			errorMatcher: nil,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+
+			result, err := Unmarshal(tc.input)
+
+			switch {
+			case err != nil && tc.errorMatcher == nil:
+				t.Fatalf("error == %#v, want nil", err)
+			case err == nil && tc.errorMatcher != nil:
+				t.Fatalf("error == nil, want non-nil")
+			case tc.errorMatcher != nil && !tc.errorMatcher(microerror.Cause(err)):
+				t.Fatalf("error == %#v, want matching", err)
+			}
+
+			if !reflect.DeepEqual(tc.matchKubeConfigValue, *result) {
+				t.Fatalf("want matching kubeconfig value \n %s", cmp.Diff(*result, tc.matchKubeConfigValue))
+			}
+		})
+	}
+}
+
+func TestKubeConfig_Marshal(t *testing.T) {
+	matchKubeConfigValue := KubeConfigValue{
+		APIVersion: "v1",
+		Kind:       "Config",
+		Clusters: []KubeconfigNamedCluster{
+			{
+				Name: "minikube",
+				Cluster: KubeconfigCluster{
+					Server:                   "https://10.142.5.51:8443",
+					CertificateAuthorityData: "Y2FkYXRhdGVzdA==",
+				},
+			},
+		},
+		Users: []KubeconfigUser{
+			{
+				Name: "minikube",
+				User: KubeconfigUserKeyPair{
+					ClientCertificateData: "Y2NkYXRhdGVzdA==",
+					ClientKeyData:         "a2V5ZGF0YXRlc3Q=",
+				},
+			},
+		},
+		Contexts: []KubeconfigNamedContext{
+			{
+				Name: "minikube",
+				Context: KubeconfigContext{
+					Cluster: "minikube",
+					User:    "minikube",
+				},
+			},
+		},
+		CurrentContext: "minikube",
+	}
+	output, err := Marshal(&matchKubeConfigValue)
+	if err != nil {
+		t.Fatalf("expect nil got %#v", err)
+	}
+
+	config, err := clientcmd.RESTConfigFromKubeConfig(output)
+	if err != nil {
+		t.Fatalf("expect nil got %#v", microerror.Mask(err))
+	}
+
+	if config.Host != "https://10.142.5.51:8443" {
+		t.Fatalf("expect config.Host same as %#v got %#v", "https://10.142.5.51:8443", config.Host)
+	}
+
+	if !bytes.Equal(config.CertData, []byte("ccdatatest")) {
+		t.Fatalf("expect config.CertData same as %#v got %#v", "ccdatatest", string(config.CertData))
+	}
+
+	if !bytes.Equal(config.CAData, []byte("cadatatest")) {
+		t.Fatalf("expect config.CAData same as %#v got %#v", "cadatatest", string(config.CAData))
+	}
+
+	if !bytes.Equal(config.KeyData, []byte("keydatatest")) {
+		t.Fatalf("expect config.CAData same as %#v got %#v", "keydatatest", string(config.CAData))
+	}
+}
+
+func TestKubeConfig_NewKubeConfigForRESTConfig(t *testing.T) {
```

Comment:
```suggestion
func Test_KubeConfig_NewKubeConfigForRESTConfig(t *testing.T) {
```

Can this be a table driven test like the others?

Reply:
Done it, thanks